src/plugins/imagery/components/ImageryViewLayout.vue

```diff
@@ -236,8 +363,180 @@ export default {
         this.openmct.time.off('bounds', this.boundsChange);
         this.openmct.time.off('timeSystem', this.timeSystemChange);
         this.openmct.time.off('clock', this.clockChange);
+
+        // unsubscribe from related telemetry
+        if (this.hasRelatedTelemetry) {
+            for (let key of this.relatedTelemetry.keys) {
+                if (this.relatedTelemetry[key].unsubscribe) {
+                    this.relatedTelemetry[key].unsubscribe();
+                }
+            }
+        }
+
+        clearInterval(this.pollResizeImageContainerID);
     },
     methods: {
+        async initializeRelatedTelemetry() {
+            if (this.imageHints.relatedTelemetry === undefined) {
+                return;
+            }
+
+            let loadedResolve;
+            this.relatedTelemetryLoaded = new Promise((resolve, reject) => {
+                loadedResolve = resolve;
+            });
+
+            let keys = Object.keys(this.imageHints.relatedTelemetry);
+
+            this.hasRelatedTelemetry = true;
+            this.relatedTelemetry = {
+                keys,
+                ...this.imageHints.relatedTelemetry
+            };
+
+            // grab historical and subscribe to realtime
+            for (let key of keys) {
+                let historicalId;
+                let realtimeId;
+
+                if (this.relatedTelemetry[key].historical) {
+                    if (this.relatedTelemetry[key].historical.telemetryObjectId) {
+                        historicalId = this.relatedTelemetry[key].historical.telemetryObjectId;
+                    } else {
+                        this.relatedTelemetry[key].historicalValuesOnTelemetry = true;
+                    }
+                }
+
+                if (this.relatedTelemetry[key].realtime && this.relatedTelemetry[key].realtime.telemetryObjectId) {
+                    realtimeId = this.relatedTelemetry[key].realtime.telemetryObjectId;
+                }
+
+                // if we have a historical object id, then values will NOT be on the imagery datum
+                if (historicalId) {
+
+                    this.relatedTelemetry[key].historicalDomainObject = await this.openmct.objects.get(historicalId);
+
+                    this.relatedTelemetry[key].requestLatestFor = async (datum) => {
+                        const options = {
+                            start: this.openmct.time.bounds().start,
+                            end: datum[this.timeKey] || datum.timestamp,
```

Comment:
We have to parse time values from telemetry data using a telemetry formatter, as we have done elsewhere in this component. The raw value could be a number, but it could also be a string. eg. YAMCS returns strings.

Reply:
Ahh.. good catch, I'll use the aptly named "parseTime" function : )